src/main/java/kotlinx/reflect/lite/misc/RuntimeTypeMapper.kt

```diff
@@ -0,0 +1,194 @@
+// Most logic copied from: https://github.com/JetBrains/kotlin/blob/master/core/reflection.jvm/src/kotlin/reflect/jvm/internal/RuntimeTypeMapper.kt
+package kotlinx.reflect.lite.misc
```

Comment:
The file looks scary: `TODO`s, much code is commented out, etc. However, as it turns out, most of the code is just unused: only a third or so of the lines are significant, and they seem fine. I think it would be better to remove the dead code, and if we ever need the other functionality, we always know where to copy-paste it from.